### PR TITLE
LUN-413: Save score on exercise finish

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/services/AsyncStorage.ts [hunk 1]
```diff
@@ -84,12 +84,38 @@ const setExerciseProgress = async (disciplineId: number, exerciseKey: ExerciseKe
   await AsyncStorage.setItem(PROGRESS_KEY, JSON.stringify(savedProgress))
 }
 
+const calculateScore = (documentsWithResults: DocumentResult[]): number => {
```
Comment (steffenkleinle) [inline src/services/AsyncStorage.ts:87]: Not sure if it is the right place here to do something like this. Imo `AsyncStorage` should be just about CRUD operations to the async storage. Not about calculating scores and similar.

File: src/services/AsyncStorage.ts [hunk 2]
```diff
@@ -84,12 +84,38 @@ const setExerciseProgress = async (disciplineId: number, exerciseKey: ExerciseKe
...
+            break
+          case 2:
+            acc += SCORE_SECOND_TRY
+            break
+          case 3:
+            acc += SCORE_THIRD_TRY
+            break
+        }
+        /* eslint-enable no-param-reassign */
+        return acc
```
Comment (steffenkleinle) [inline src/services/AsyncStorage.ts:108]:
```suggestion
        let score = acc
        switch (document.numberOfTries) {
          case 1:
            score += SCORE_FIRST_TRY
            break
          case 2:
            score += SCORE_SECOND_TRY
            break
          case 3:
            score += SCORE_THIRD_TRY
            break
        }
        return score
```

**Review by steffenkleinle — APPROVED** [review]

File: src/services/AsyncStorage.ts (hunk 1 above)
Comment (SteffiStuffel) [reply src/services/AsyncStorage.ts:87]: moved to helpers

**Review by f1sh1918 — APPROVED** [review]
tested on ios. calculation works fine! nice work :)

**Pull request merged**